java/sage/io/SageFileSource.java

```diff
@@ -0,0 +1,277 @@
+   */
+  public void randomWrite(long pos, byte b[], int off, int len) throws IOException;
+
+  /**
+   * Skip a number of bytes.
+   * <p/>
+   * Values less than 0 will return 0 bytes skipped. If the value of <code>n</code> moves the
+   * position in the file past the end of the file, the position will be set to the length of the
+   * file.
+   *
+   * @param n The number of bytes to skip.
+   * @return The number of bytes actually skipped.
+   * @throws IOException If there is an I/O related error.
+   */
+  public long skip(long n) throws IOException;
+
+  /**
+   * Seek to a specific position relative to the start of the file.
+   *
+   * @param pos The new file position.
+   * @throws IOException If there is an I/O related error or <code>pos</code> is negative.
+   */
+  public void seek(long pos) throws IOException;
+
+  /**
+   * Get the position from which the next byte will be read or written relative to the start of the
+   * file.
+   * <p/>
+   * This method may return -1 if getting the position causes an I/O exception.
+   *
+   * @return The current position.
+   */
+  public long position();
+
+  /**
+   * Get the current length of the file.
+   *
+   * @return The current length of the file.
+   * @throws IOException If there is an I/O related error.
+   */
+  public long length() throws IOException;
+
+  /**
+   * Get the number of bytes remaining in the file.
+   *
+   * @return The number of bytes remaining.
+   * @throws IOException If there is an I/O related error.
+   */
+  public long available() throws IOException;
+
+  /**
+   * Changes the current length of the file.
+   * <p/>
+   * If the current file position is greater than the new length, the file position will be changed
+   * to equal the new file length. If the new file length is greater than the current size of the
+   * file, the file will grow to the new size.
+   *
+   * @param newLength The new file length.
+   * @throws IOException If there is an I/O related error, the file was opened as read-only or this
+   *                     implementation does not support writing.
+   */
+  public void setLength(long newLength) throws IOException;;
+
+  /**
+   * Sync data to disk.
+   *
+   * @throws IOException If there is an I/O related error.
+   */
+  public void sync() throws IOException;
+
+  /**
+   * Flush buffers to disk.
+   * <p/>
+   * This will generally have no effect unless there is a buffer layer present.
+   *
+   * @throws IOException If there is an I/O related error.
+   */
+  public void flush() throws IOException;
+
+  /**
+   * Close file.
+   * <p/>
+   * Ensure that any form of buffering (if any) is flushed before closing the file.
+   *
+   * @throws IOException If there is an I/O related error.
+   */
+  public void close() throws IOException;
+
+  /**
+   * Is this file actively growing?
+   *
+   * @return <code>true</code> if the file is actively growing.
+   */
+  public boolean isActiveFile();
```

Comment:
I don't think you should have the isActiveFile() method in here. That's a much higher level concept, and it's also not actually implemented except for network files so I don't want people thinking they can use it. There is MMC.isRecording(File) which will give you the correct answer...BUT that does not work properly unless you are in the server process (i.e. there's no way to call this on SageTVClient, but yes, the client should be accessing files remotely, but for the service case there may be optimizations where it is not).

As I also don't see any actual usages of this yet in the code, removing it doesn't seem like a problem.